XS.pm

```diff
@@ -484,35 +484,37 @@ non-JSON conformant hex or octal encoding C<\xNN> or C<\NNN>.
   Error: illegal hex character in non-binary string
 
 
-=item $json = $json->utf8 ([$enable])
+=item $json = $json->utf8 ([$mode_constant])
 
-=item $enabled = $json->get_utf8
+=item $mode_constant = $json->get_utf8
 
-If C<$enable> is true (or missing), then the C<encode> method will encode
-the JSON result into UTF-8, as required by many protocols, while the
-C<decode> method expects to be handled an UTF-8-encoded string.  Please
-note that UTF-8-encoded strings do not contain any characters outside the
-range C<0..255>, they are thus useful for bytewise/binary I/O. In future
-versions, enabling this option might enable autodetection of the UTF-16
-and UTF-32 encoding families, as described in RFC4627.
+Determines how $json handles UTF-8 encoding and decoding.
+
+$mode_constant is one of the following constants under Cpanel::JSON::XS:
 
-If C<$enable> is false, then the C<encode> method will return the JSON
+=over
+
+=item * B<UTF8_CHARS> - The C<encode> method will return the JSON
 string as a (non-encoded) Unicode string, while C<decode> expects thus a
 Unicode string.  Any decoding or encoding (e.g. to UTF-8 or UTF-16) needs
 to be done yourself, e.g. using the Encode module.
 
-See also the section I<ENCODING/CODESET FLAG NOTES> later in this
-document.
-
-Example, output UTF-16BE-encoded JSON:
+=item * C<UTF8_STANDARD> - The default if no $mode_constant is given.
+The C<encode> method will encode
+the JSON result into UTF-8, as required by many protocols, while the
+C<decode> method expects to be handed a UTF-8-encoded string.  Please
+note that UTF-8-encoded strings do not contain any characters outside the
+range C<0..255>; they are thus useful for bytewise/binary I/O.
 
-  use Encode;
-  $jsontext = encode "UTF-16BE", Cpanel::JSON::XS->new->encode ($object);
+=item * C<UTF8_BYTES> - Like B<UTF8_CHARS>, but C<encode> and C<decode>
+expect and return UTF-8 strings rather than Unicode strings. This mode
```

Comment:
This describes the operation currently implemented by utf8(1). The only reasonable operation on UTF-8 strings is to decode them from UTF-8 because of the existence of non-encoded escapes.